lib/gis_robot_suite/arcgis_metadata_transformer.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module GisRobotSuite
+  class ArcgisMetadataTransformer
+    VALID_FORMATS = {
+      'ISO19139' => {
+        xslt: 'ArcGIS2ISO19139.xsl',
+        output: 'iso19139.xml'
+      },
+      'ISO19110' => {
+        xslt: 'arcgis_to_iso19110.xsl',
+        output: 'iso19110.xml'
+      },
+      'FGDC' => {
+        xslt: 'ArcGIS2FGDC.xsl',
+        output: 'fgdc.xml'
+      }
+    }.freeze
+
+    def self.transform(druid, format, logger: nil)
+      new(druid, format, logger).transform
+    end
+
+    def initialize(druid, format, logger)
+      @druid = druid
+      @format = format.upcase
+      @logger = logger
+
+      raise "ArcgisMetadataTransformer: #{format} is not a valid format" unless VALID_FORMATS.key?(format)
+    end
+
+    def transform
+      logger&.info("extract-#{format}-metadata: generating #{output_file} using #{xslt_file}")
+      return if format == 'ISO19110' && data_type != 'Shapefile'
+
+      system("#{xslt_command} #{xslt_file} '#{esri_metadata_file}' | #{xml_lint_command} -o '#{output_file}' -", exception: true)
+    end
+
+    attr_reader :druid, :format, :logger
+
+    private
+
+    def xslt_file
+      File.join(xslt_path, VALID_FORMATS[format][:xslt]) if VALID_FORMATS.key?(format)
+    end
+
+    def output_file
+      File.join(staging_dir, 'temp', "#{layer_name}-#{VALID_FORMATS[format][:output]}") if VALID_FORMATS.key?(format)
+    end
```

Comment:
do we need the conditional checks here if we have one in `initialize` too?

Reply:
Probably note, semi-cruft